core/kt-osrd-path/src/main/kotlin/fr/sncf/osrd/path/interfaces/GenericLinearRange.kt

```diff
@@ -25,39 +24,54 @@ data class GenericLinearRange<ValueType, OffsetType>(
     val value: ValueType,
     /** Start of the range, compared to the start of the underlying object. */
     val objectBegin: Offset<OffsetType>,
-    /** End of the range, compared to the start of the underlying object. */
-    val objectEnd: Offset<OffsetType>,
     /** Start of the range, compared to the start of the train path. */
     val pathBegin: Offset<TrainPath>,
-    /** End of the range, compared to the start of the train path. */
-    val pathEnd: Offset<TrainPath>,
+    /** Range length. */
+    val length: Distance,
+    /** Total length of the underlying object. */
+    val objectLength: Length<OffsetType>,
 ) {
-    val length = objectEnd - objectBegin
+    constructor(
+        value: ValueType,
+        objectBegin: Offset<OffsetType>,
+        objectEnd: Offset<OffsetType>,
+        pathBegin: Offset<TrainPath>,
+        pathEnd: Offset<TrainPath>,
+        objectLength: Length<OffsetType>,
+    ) : this(value, objectBegin, pathBegin, pathEnd - pathBegin, objectLength) {
+        assert(objectEnd - objectBegin == pathEnd - pathBegin)
+    }
 
     init {
         require(length >= 0.meters)
-        require(pathEnd - pathBegin == length)
+        require(objectEnd <= objectLength)
```

Comment:
Is the range always within the object? Why can `objectBegin` be negative but `objectEnd` cannot go past the object's length? just asking, since it doesn't seem we build a range with `objectBegin < 0` in tests